db/models.py

```diff
@@ -50,3 +55,49 @@ class MovieSession(models.Model):
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
+
+    class Meta:
+        ordering = ["-created_at"]
+
+    def __str__(self) -> str:
+        return f"Order: {self.created_at}"
+
+
+class Ticket(models.Model):
+    movie_session = models.ForeignKey(to=MovieSession, on_delete=models.CASCADE)
+    order = models.ForeignKey(to=Order, on_delete=models.CASCADE)
+    row = models.IntegerField()
+    seat = models.IntegerField()
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=["movie_session", "seat", "row"],
+                name="unique_ticket",
+            )
+        ]
+
+    def clean(self):
+        if self.row <= 0 and self.row > self.movie_session.cinema_hall.rows:
```

Comment:
The condition `self.row <= 0 and self.row > self.movie_session.cinema_hall.rows` seems incorrect. It should likely be `self.row <= 0 or self.row > self.movie_session.cinema_hall.rows` to properly validate that the row number is within the valid range.